test/unit/CowWrapper.t.sol

```diff
@@ -0,0 +1,160 @@
+        return
+            [
+                new ICowSettlement.Interaction[](0),
+                new ICowSettlement.Interaction[](0),
+                new ICowSettlement.Interaction[](0)
+            ];
+    }
+
+    function _createSimpleSettleData(uint256 tokenCount) private returns (bytes memory) {
+        address[] memory tokens = new address[](tokenCount);
+        uint256[] memory clearingPrices = new uint256[](tokenCount);
+        for (uint256 i = 0; i < tokenCount; i++) {
+            tokens[i] = makeAddr(string(abi.encodePacked("Settle Token #", vm.toString(i + 1))));
+            clearingPrices[i] = 100 * (i + 1);
+        }
+        return abi.encodeWithSelector(
+            ICowSettlement.settle.selector, tokens, clearingPrices, new ICowSettlement.Trade[](0), _emptyInteractions()
+        );
+    }
+
+    function test_next_CallsWrapperAndThenNextSettlement() public {
+        bytes memory settleData = abi.encodePacked(_createSimpleSettleData(1), hex"123456");
+        bytes memory secondCallWrapperData = hex"0003098765";
+        bytes memory wrapperData = abi.encodePacked(hex"00021234", address(wrapper1), secondCallWrapperData);
+
+        // the wrapper gets called exactly twice (once below and again inside the wrapper data calling self)
+        vm.expectCall(address(wrapper1), 0, abi.encodeWithSelector(wrapper1.wrappedSettle.selector), 2);
+
+        // verify the internal wrapper call data
+        vm.expectCall(
+            address(wrapper1),
+            abi.encodeWithSelector(wrapper1.wrappedSettle.selector, settleData, secondCallWrapperData)
+        );
+
+        // the settlement contract gets called once after wrappers (including the surplus data at the end)
+        vm.expectCall(address(mockSettlement), 0, settleData, 1);
+
+        vm.prank(solver);
+        wrapper1.wrappedSettle(settleData, wrapperData);
+    }
+
+    function test_wrappedSettle_RevertsWithNotASolver() public {
+        bytes memory settleData = _createSimpleSettleData(0);
+        address notASolver = makeAddr("notASolver");
+
+        // Should revert when called by non-solver
+        vm.prank(notASolver);
+        vm.expectRevert(abi.encodeWithSelector(CowWrapper.NotASolver.selector, notASolver));
+        wrapper1.wrappedSettle(settleData, hex"");
+    }
+
+    function test_wrappedSettle_RevertsOnInvalidSettleSelector() public {
+        bytes memory settleData = abi.encodePacked(bytes4(0xdeadbeef), hex"1234");
+        bytes memory wrapperData = hex"0000"; // Empty wrapper data, goes straight to settlement
+        vm.prank(solver);
+        vm.expectRevert(abi.encodeWithSelector(CowWrapper.InvalidSettleData.selector, settleData));
+        wrapper1.wrappedSettle(settleData, wrapperData);
+    }
+
+    function test_integration_ThreeWrappersChained() public {
+        // Set up a more sophisticated settlement call to make sure it all gets through as expected.
```

Comment:
Is it needed? I think the wrapper doesn't care about what the settle data is apart for the selector, so for a unit test all of this seems unnecessary and could be simplified. For an e2e test this would be reasonable though.